File: backend/app/schemas/produtos.py
from pydantic import Field
from app.schemas.common import ORMBase, TimestampSchema


class ProdutoBase(ORMBase):
    nome: str = Field(min_length=1, max_length=120)
    categoria: str = Field(pattern="^(renda_fixa|previdencia|fgts|fundo|outro)$")
    instituicao_id: int
    moeda: str = Field(default="BRL", pattern="^(BRL|USD)$")


class ProdutoCreate(ProdutoBase):
    pass


class ProdutoUpdate(ORMBase):
    nome: str | None = None
    categoria: str | None = None
    instituicao_id: int | None = None
    moeda: str | None = None
    ativo: int | None = None


class ProdutoOut(ProdutoBase, TimestampSchema):
    id: int
    ativo: int

File: frontend/src/services/produtos.ts
import { http } from "./http"

export interface ProdutoInvestimento {
  id: number
  nome: string
  categoria: "renda_fixa" | "previdencia" | "fgts" | "fundo" | "outro"
  instituicao_id: number
  moeda: "BRL" | "USD"
  ativo: number
  criado_em: string
  atualizado_em: string
}

export const produtosService = {
  listar: (apenasAtivos = false) =>
    http.get<ProdutoInvestimento[]>("/produtos-investimento",
      { params: { apenas_ativos: apenasAtivos } }).then(r => r.data),
  criar: (p: Partial<ProdutoInvestimento>) =>
    http.post<ProdutoInvestimento>("/produtos-investimento", p).then(r => r.data),
  atualizar: (id: number, p: Partial<ProdutoInvestimento>) =>
    http.put<ProdutoInvestimento>(`/produtos-investimento/${id}`, p).then(r => r.data),
  inativar: (id: number) => http.delete(`/produtos-investimento/${id}`),
}